### PR TITLE
Add JNI for `lists::concatenate_list_elements`

### PR DESCRIPTION
**Pull request by ttnghia**
This implements JNI work for `lists::concatenate_list_elements` to expose the API to Java usage.

### CODE REVIEW

File: java/src/main/native/src/ColumnVectorJni.cpp [hunk 1]
```diff
@@ -214,6 +214,22 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_concatListByRow(JNIEnv
...
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_flattenLists(JNIEnv *env, jclass,
+                                                                      jlong input_handle,
+                                                                      jboolean ignore_null) {
+  JNI_NULL_CHECK(env, input_handle, "input_handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    auto const null_policy = ignore_null ? cudf::lists::concatenate_null_policy::IGNORE :
+                                           cudf::lists::concatenate_null_policy::NULLIFY_OUTPUT_ROW;
+
```
Comment (ttnghia) [inline java/src/main/native/src/ColumnVectorJni.cpp:225]:
```suggestion
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 2]
```diff
@@ -2491,6 +2491,31 @@ public final ColumnVector dropListDuplicatesWithKeysValues() {
...
+  public ColumnVector flattenLists() {
+    return flattenLists(false);
+  }
+
+  /**
+   * Flatten each list of lists into a single list.
+   *
+   * The column must have rows that are lists of lists.
+   *
+   * @param ignoreNull Whether to ignore null list elements of input columns from the operation,
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:2511]:
```suggestion
   * @param ignoreNull Whether to ignore null list elements in the input column from the operation,
```

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 3]
```diff
@@ -2876,6 +2876,23 @@ void testListConcatByRowIgnoreNull() {
...
+  @Test
+  void testFlattenLists() {
+    HostColumnVector.ListType listType = new HostColumnVector.ListType(true,
+        new HostColumnVector.BasicType(true, DType.INT32));
+    HostColumnVector.ListType listOfListsType = new HostColumnVector.ListType(true, listType);
+
+    try (ColumnVector input = ColumnVector.fromLists(listOfListsType,
+             Arrays.asList(Arrays.asList(1, 2), Arrays.asList(3), Arrays.asList(4, 5, 6)),
+             Arrays.asList(Arrays.asList(7, 8, 9), Arrays.asList(10, 11, 12, 13, 14, 15)));
+         ColumnVector result = input.flattenLists();
```
Comment (tgravescs) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:2888]: do we have a test for flattenLists(true) - to ignoreNulls ?

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 4]
```diff
@@ -2491,6 +2491,31 @@ public final ColumnVector dropListDuplicatesWithKeysValues() {
...
+  public ColumnVector flattenLists() {
+    return flattenLists(false);
+  }
+
+  /**
+   * Flatten each list of lists into a single list.
+   *
+   * The column must have rows that are lists of lists.
+   *
+   * @param ignoreNull Whether to ignore null list elements in the input column from the operation,
```
[the remark this answers is not in the document]
Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:2511]: No, that means "do not look at these nulls, just move on without doing anything with them".

[the remark this answers is not in the document]
Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:2511]: Yes an example would be good.

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java (hunk 3 above)
Comment (ttnghia) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:2888]: Sure we can have such test.

**Review by revans2 — APPROVED** [review]
Looks good to me. Just have a nit with the docs, but beyond that it is fine.

**Pull request merged**